.github/workflows/ci.yml

```diff
@@ -79,11 +80,27 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all commits of this PR except HEAD compile
+        run: |
+          git reset --hard HEAD~1
+          git rebase --exec "$MAVEN test $MAVEN_COMPILE_COMMITS" refs/remotes/origin/${{ github.event.pull_request.base.ref }}
```

Comment:
Can you show an example of failure for a non-head commit? Would it be easy enough to identify which commit failed?

Reply:
I'll prepare a fake commit on top of this one which will fail.

Reply:
```
Error:  Failed to execute goal io.trino:trino-maven-plugin:11:generate-service-descriptor (default-generate-service-descriptor) on project trino-hive-hadoop2: Execution default-generate-service-descriptor of goal io.trino:trino-maven-plugin:11:generate-service-descriptor failed: A required class was missing while executing io.trino:trino-maven-plugin:11:generate-service-descriptor: io/trino/plugin/hive/HiveConnectorFactory
Error:  -----------------------------------------------------
Error:  realm =    extension>io.trino:trino-maven-plugin:11
Error:  strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
Error:  urls[0] = file:/home/runner/.m2/repository/io/trino/trino-maven-plugin/11/trino-maven-plugin-11.jar
Error:  urls[1] = file:/home/runner/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.17/plexus-utils-3.0.17.jar
Error:  Number of foreign imports: 1
Error:  import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
Error:  
Error:  -----------------------------------------------------: io.trino.plugin.hive.HiveConnectorFactory
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :trino-hive-hadoop2
interactive rebase in progress; onto daa858da30
Last commands done (6 commands done):
   pick 3fc5c79e20 Intentionally break a plugin
   exec ./mvnw package -B --quiet --strict-checksums --no-snapshot-updates --no-transfer-progress -T C1 -DskipTests -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -Dair.check.skip-all=true -P gib -Dgib.referenceBranch=refs/remotes/origin/master -pl '!:trino-server-rpm' || (git status && false)
  (see more in file .git/rebase-merge/done)
No commands remaining.
You are currently editing a commit during a rebase.
  (use "git commit --amend" to amend the current commit)
  (use "git rebase --continue" once you are satisfied with your changes)

nothing to commit, working tree clean
warning: execution failed: ./mvnw package -B --quiet --strict-checksums --no-snapshot-updates --no-transfer-progress -T C1 -DskipTests -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -Dair.check.skip-all=true -P gib -Dgib.referenceBranch=refs/remotes/origin/master -pl '!:trino-server-rpm' || (git status && false)
You can fix the problem, and then run

  git rebase --continue


```